zerver/webhooks/splunk/view.py

```diff
@@ -0,0 +1,37 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from typing import Dict, Any, Iterable, Optional, Text
+
+@api_key_only_webhook_view('Splunk')
+@has_request_variables
+def api_splunk_webhook(request, user_profile, client,
+                       payload=REQ(argument_type='body'), stream=REQ(default='splunk'),
+                       topic=REQ(default='Splunk Alert')):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Any], Text, Optional[Text]) -> HttpResponse
+
+    # use default values if expected data is not provided
+    search_name = payload.get('search_name', 'Missing search_name')
+    results_link = payload.get('results_link', 'Missing results_link')
+    host = payload.get('result', {}).get('host', 'Missing host')
+    source = payload.get('result', {}).get('source', 'Missing source')
+    raw = payload.get('result', {}).get('_raw', 'Missing _raw')
```

Comment:
Structurally, I'm wondering if we can use topics a bit more effectively; e.g. rather than having the user control the topic when setting up the webhook, using the host or source or specific search or whatever in the topic?  That would have the nice effective of grouping related things together so that one can just read a topic to see the history of something.  Our git integration is a good example of doing this well.

Also are "saved search"es the only source of alerts?